applications/GeoMechanicsApplication/custom_utilities/check_utilities.h

```diff
@@ -39,4 +40,136 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) CheckUtilities
+    }
+
+    template <typename T>
+    void CheckAvailabilityOnly(const Variable<T>& rVariable) const
+    {
+        if (!mrProperties.Has(rVariable))
+            KRATOS_ERROR << rVariable.Name() << " does not exist in the " << mrPrintName << " "
+                         << mId << "." << std::endl;
+    }
+
+    template <typename T>
+    void CheckAvailabilityAndEmpty(const Variable<T>& rVariable) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        if (mrProperties[rVariable].empty())
+            KRATOS_ERROR << rVariable.Name() << " is empty in the " << mrPrintName << " " << mId
+                         << "." << std::endl;
+    }
+
+    void CheckPermeabilityProperties(size_t Dimension) const;
+
+private:
+    const Properties&           mrProperties;
+    const std::string           mrPrintName;
+    const std::size_t           mId;
+    mutable Bounds              mRangeBoundsType;
+    const double                mDefaultLowerBound = 0.0;
+    const double                mDefaultUpperBound = std::numeric_limits<double>::max();
+    mutable std::vector<Bounds> mHistoryOfTypes;
+
+    template <typename T>
+    void CheckRangeBounds(const Variable<T>& rVariable, double LowerBound, double UpperBound) const
+    {
+        const auto value = mrProperties[rVariable];
+        bool       in_range;
+        using enum CheckProperties::Bounds;
+        switch (mRangeBoundsType) {
+        case AllExclusive:
+            in_range = (value > LowerBound && value < UpperBound);
+            break;
+        case AllInclusive:
+            in_range = (value >= LowerBound && value <= UpperBound);
+            break;
+        case InclusiveLowerAndExclusiveUpper:
+            in_range = (value >= LowerBound && value < UpperBound);
+            break;
+        case ExclusiveLowerAndInclusiveUpper:
+            in_range = (value > LowerBound && value <= UpperBound);
+            break;
+        default:
+            KRATOS_ERROR << " Unknown type of range bounds";
+            break;
+        }
+        if (!in_range) {
+            std::ostringstream print_range;
+            const auto         include_lower_bound = (mRangeBoundsType == AllInclusive) ||
+                                             (mRangeBoundsType == InclusiveLowerAndExclusiveUpper);
+            const auto include_upper_bound = (mRangeBoundsType == AllInclusive) ||
+                                             (mRangeBoundsType == ExclusiveLowerAndInclusiveUpper);
+            print_range << (include_lower_bound ? "[" : "(") << LowerBound << "; "
+                        << ((UpperBound == std::numeric_limits<double>::max()) ? "-" : std::to_string(UpperBound))
+                        << (include_upper_bound ? "]" : ")");
+            KRATOS_ERROR << rVariable.Name() << " in the " << mrPrintName << " " << mId
+                         << " has an invalid value: " << value << " out of the range "
+                         << print_range.str() << "." << std::endl;
```

Comment:
I'd propose this slight rewording:
```suggestion
            KRATOS_ERROR << rVariable.Name() << " in the " << mrPrintName << " with Id" << mId
                         << " has an invalid value: " << value << " is out of the range "
                         << print_range.str() << "." << std::endl;
```

Reply:
changed